tee-worker/litentry/primitives/src/lib.rs

```diff
@@ -108,23 +111,39 @@ impl LitentryMultiSignature {
 	fn verify_evm(&self, msg: &[u8], signer: &Address20) -> bool {
 		match (self, signer) {
 			(Self::Ethereum(ref sig), who) => {
-				let digest = keccak_256(msg);
-				return match recover_evm_address(&digest, sig.as_ref()) {
-					Ok(recovered_evm_address) => recovered_evm_address == who.as_ref().as_slice(),
-					Err(_e) => {
-						error!(
-							"Could not verify evm signature msg: {:?}, signer {:?}",
-							msg, signer
-						);
-						false
-					},
-				}
+				let data = msg;
+				return verify_evm_signature(evm_eip191_wrap(data).as_slice(), sig, who, msg, signer)
+					|| verify_evm_signature(data, sig, who, msg, signer)
+			},
+			(Self::EthereumPrettified(ref sig), who) => {
+				let user_readable_message =
+					"Litentry authorization token: 0x".to_string() + &hex::encode(msg);
```

Comment:
Maybe we can consider using this: https://github.com/litentry/litentry-parachain/blob/aa12180696bb51505d98415aa32e711b7275f131/tee-worker/core-primitives/utils/src/hex.rs#L52

essentially the same but we try to keep the source of truth consistent

Reply:
Tried that earlier but couldn't get it to work for some reason; gave it another go and it worked like a charm this time :raised_hands: Thanks!